### ISSUE TITLE
[warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in [from nginx 1.25.1 onwards]

### ISSUE THREAD

**Issue by xepa** [voxpupuli/puppet-nginx#1593]
In version 1.25.1 and later http2 directive in listen statement is deprecated, the current method should only change operations if the version of nginx is 1.25.1 or later (older versions MUST retain this method).

For reference please see https://nginx.org/en/docs/http/ngx_http_v2_module.html#http2

## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: all
- Ruby: all
- Distribution: all
- Module version: 

## How to reproduce (e.g Puppet code you use)
```puppet
nginx::resource::server { 'example.com':
  ssl            => true,
  http2       => true,
  ... and others ...
}
```
## What are you seeing
```
# MANAGED BY PUPPET
server {
  listen       *:443 ssl http2;
  ... and others ...
}
```

## What behaviour did you expect instead
```
# MANAGED BY PUPPET
server {
  listen       *:443 ssl;
  http2        on;
  ... and others ...
}
```

## Output log
```
2024/05/08 12:34:46 [warn] 2465396#2465396: the "listen ... http2" directive is deprecated, use the "http2" directive instead in /etc/nginx/sites-enabled/100-sentry.veaplatform.nl_https.conf:4
```

**Comment by TheMeier** [comment]
Hm isn't that already implemented

https://github.com/voxpupuli/puppet-nginx/blob/7f3a8ae33c576c190852ce0a8c7cae6ab929336c/templates/server/server_ssl_settings.erb#L4-L6

**Comment by TheMeier** [comment]
I guess the issue is, last release was  Jun 26, 2023

**Comment by xepa** [comment]
I was not aware that this was even an option, not sure why my code is even doing this I will investigate .. I will reopen if I can not get this to work ..

**Comment by TheMeier** [comment]
As I pointed out, that feature is not released :/

**Comment by 999faryad** [comment]
Problem still occurs in the latest version. also with     
`http2 => 'on',`

**Comment by 999faryad** [comment]
@xepa since you have not reopened the issue, have you found a workaround / fix?

**Comment by kenyon** [comment]
@999faryad it was fixed in #1579, but there hasn't been a release containing this yet. You can use the module from git instead of a release.

**Comment by xepa** [comment]
> @xepa since you have not reopened the issue, have you found a workaround / fix?
I was using the master ..
